netutils/password.py

```diff
@@ -197,23 +142,37 @@ def encrypt_type7(unencrypted_password, salt=None):
         salt (str, optional): A random number between 0 and 15 that can be set by the operator. Defaults to random generated one.
 
     Returns:
-        string: The encrypted password.
+        string (optional): The encrypted password.
 
     Example:
         >>> from netutils.password import encrypt_type7
-        >>> encrypt_type5("cisco")  # doctest: +SKIP
-        '$1$ZLGo$J.gAGxS2wqO96drs0Cith/'
+        >>> encrypt_type7("cisco", 11)
+        '110A1016141D'
         >>>
     """
-    if not salt:
-        salt = random.randrange(0, 15)  # nosec
-    encrypted_password = "%02x" % salt  # pylint: disable=consider-using-f-string
-    for i, _ in enumerate(unencrypted_password):
-        hex_password = "%02x" % (ord(unencrypted_password[i]) ^ XLAT[salt])  # pylint: disable=consider-using-f-string
-        encrypted_password += hex_password
-        salt += 1
-        if salt == 51:
-            salt = 0
+    encrypted_password = ""  # nosec
+    # max length of password for encrypt t7 is 25
+    if len(unencrypted_password) <= 25:  # nosec
```

Comment:
Generally speaking, I try to limit nesting of configurations, this is a pattern to help with that, as the rest can be de-indented. 
```suggestion
    if len(unencrypted_password) > 25:  # nosec
        raise ValueError("Password must not exceed 25 characters.")
```